src/hotspot/share/prims/agent.hpp

```diff
@@ -0,0 +1,88 @@
+/*
```

Comment:
The name for class and file is too general.
I'm thinking if renaming the files to jvmtiAgent and the class to JvmtiAgent would work.
In general, there exists a convention to name JVMTI file with the "jvmti" prefix.
It is a gray zone between Runtime and JVMTI but seems to belong more to JVMTI.
The same about the AgentList class and file.
Also, these new files are good candidates to add here:
```
make/hotspot/lib/JvmFeatures.gmk:
ifneq ($(call check-jvm-feature, jvmti), true)
  JVM_CFLAGS_FEATURES += -DINCLUDE_JVMTI=0
  JVM_EXCLUDE_FILES += jvmtiGetLoadedClasses.cpp jvmtiThreadState.cpp jvmtiExtensions.cpp \
      jvmtiImpl.cpp jvmtiManageCapabilities.cpp jvmtiRawMonitor.cpp jvmtiUtil.cpp jvmtiTrace.cpp \
      jvmtiCodeBlobEvents.cpp jvmtiEnv.cpp jvmtiRedefineClasses.cpp jvmtiEnvBase.cpp jvmtiEnvThreadState.cpp \
      jvmtiTagMap.cpp jvmtiEventController.cpp evmCompat.cpp jvmtiEnter.xsl jvmtiExport.cpp \
      jvmtiClassFileReconstituter.cpp jvmtiTagMapTable.cpp
endif
```

Reply:
Thank you for the update. It looks good to me.
I still need to finish my review. Sorry for the latency. It is is a busy time now.